pkg/kubelet/rkt/rkt.go

```diff
@@ -142,3 +148,248 @@ func (r *Runtime) runCommand(args ...string) ([]string, error) {
 	}
 	return strings.Split(strings.TrimSpace(string(output)), "\n"), nil
 }
+
+type resource struct {
+	limit   string
+	request string
+}
+
+// setIsolators overrides the isolators of the pod manifest if necessary.
+func setIsolators(app *appctypes.App, c *api.Container) error {
+	var isolator appctypes.Isolator
+	if len(c.Capabilities.Add) > 0 || len(c.Capabilities.Drop) > 0 || len(c.Resources.Limits) > 0 || len(c.Resources.Requests) > 0 {
+		app.Isolators = []appctypes.Isolator{}
+	}
+
+	// Retained capabilities/privileged.
+	privileged := false
+	if capabilities.Get().AllowPrivileged {
+		privileged = c.Privileged
+	} else if c.Privileged {
+		glog.Errorf("rkt: Privileged is disallowed globally")
+		// TODO(yifan): Return error?
+	}
+	var addCaps string
+	if privileged {
+		addCaps = getAllCapabilities()
+	} else {
+		addCaps = getCapabilities(c.Capabilities.Add)
+	}
+	if len(addCaps) > 0 {
+		// TODO(yifan): Replace with constructor, see:
+		// https://github.com/appc/spec/issues/268
+		value := []byte(fmt.Sprintf(`{"name":"os/linux/capabilities-retain-set","value":{"set":[%s]}`, addCaps))
+		if err := isolator.UnmarshalJSON(value); err != nil {
+			glog.Errorf("rkt: Cannot unmarshal the retained capabilites %q: %v", value, err)
+			return err
+		}
+		app.Isolators = append(app.Isolators, isolator)
+	}
+
+	// Removed capabilities.
+	dropCaps := getCapabilities(c.Capabilities.Drop)
+	if len(dropCaps) > 0 {
+		// TODO(yifan): Replace with constructor, see:
+		// https://github.com/appc/spec/issues/268
+		value := []byte(fmt.Sprintf(`{"name":"os/linux/capabilities-remove-set","value":{"set":[%s]}`, dropCaps))
+		if err := isolator.UnmarshalJSON(value); err != nil {
+			glog.Errorf("rkt: Cannot unmarshal the retained capabilites %q: %v", value, err)
+			return err
+		}
+		app.Isolators = append(app.Isolators, isolator)
+	}
+
+	// Resources.
+	resources := make(map[api.ResourceName]resource)
```

Comment:
Since resource is not a pointer we can't edit it in place. We're modifying a copy:

http://play.golang.org/p/eYtNFuq5N8